modules/serverless/pages/serving/serving-transport-encryption.adoc

```diff
@@ -0,0 +1,446 @@
+
+[IMPORTANT]
+====
+{serverlessproductname} transport encryption for Serving is a Developer Preview feature only.
+Developer Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete.
+Red Hat does not recommend using them in production.
+These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.
+
+For more information about the support scope of Red Hat Developer Preview features, see https://access.redhat.com/support/offerings/devpreview/.
+====
+
+
+
+== Overview
+
+There are three parts to {serverlessproductname} Serving Transport Encryption:
+
+image::serving-encryption/encryption-overview.drawio.svg[]
+
+1. Transport Encryption on the ingress layer external to the cluster (e.g. cluster external domain, like `myapp-<namespace>.example.com`).
+2. Transport Encryption on the ingress layer internal to the cluster (e.g. cluster local domains, like `myapp.<namespace>.svc.cluster.local`).
+3. Transport Encryption between Knative internal components (ingress-controller, activator, queue-proxy).
+
+[IMPORTANT]
+====
+Currently, all control-plane traffic (including Kubernetes PreStopHooks, metadata and metrics) is not encrypted.
+====
+
+=== The parts in detail
+
+The different parts are independent of each other and can be enabled/disabled individually. Also, they can use the same or different Certificate Authorities (CAs) to sign the necessary certificates.
+
+==== External domain encryption
+
+The transport encryption for external domains is handled by the cluster's ingress layer.
+This is either {product-title} ingress or {smproductname}.
+Please refer to the relevant documentation for more information.
+
+==== Cluster-local encryption
+
+image::serving-encryption/encryption-cluster-local-domain.drawio.svg[]
+
+This part enables transport encryption for cluster-local domains. It has the following properties:
+
+* Certificate CN/SAN contains the cluster-local domains of a Knative Service, e.g. `myapp.namespace.svc.cluster.local`, `myapp.namespace.svc`, `myapp.namespace`.
+* The certificates are hosted using SNI by the cluster-local endpoint of the ingress-controller.
+* The caller has to trust the CA that signed the certificates (this is out of the scope of {ServerlessProductName}).
+* To create the certificates, Knative relies on cert-manager. cert-manager needs to be installed and configured for the feature to work.
+
+==== {ServerlessProductName} Serving system-internal encryption
+
+image::serving-encryption/encryption-system-internal.drawio.svg[]
+
+This part enables transport encryption for system internal components (Ingress-Controller, Activator, Queue-Proxy). These components are hosting TLS endpoints when this configuration is used:
```

Comment:
Same comment as before. Should this be "Ingress-Controller" or "Ingress gateway". Maybe I'm too picky here but I would like to understand this.